cypress/integration/Policy/policy.js

```diff
@@ -0,0 +1,181 @@
+/**
+ * Copyright 2021 The Rode Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Before, Given, When, Then } from "cypress-cucumber-preprocessor/steps";
+import * as selectors from "../../page-objects/policy";
+import policies from "../../fixtures/policies.json";
+import Chance from "chance";
+import {
+  mockFailedPatchPolicyResponse,
+  mockFailedPolicyValidation,
+  mockSuccessPolicyValidation,
+} from "../../mock-responses/policy-responses";
+import { capitalize } from "../../page-objects/utils";
+
+const chance = new Chance();
+
+let updatedValues;
+
+Before({ tags: "@updatePolicy" }, () => {
+  updatedValues = {
+    name: chance.string(),
+    description: chance.sentence(),
+    regoContent:
+      'package testPackage\npass = true\n\nviolations[result] {\n    result:={\n        "pass": true\n        }\n}',
+  };
+});
+
+Given(/^I am on the "([^"]*)" policy details page$/, (policyName) => {
+  cy.mockRequest(
+    { url: "**/api/policies/*", method: "GET" },
+    policies[policyName]
+  );
+  cy.visit(`/policies/${policies[policyName].id}`);
+});
+
+When(/^I test invalid rego policy code$/, () => {
+  cy.mockRequest(
+    {
+      url: "**/api/policies/validate",
+      method: "POST",
+    },
+    mockFailedPolicyValidation
+  );
+
+  cy.get(selectors.PolicyRegoContentInput)
+    .clear()
+    .type(`Package ${chance.word()}`);
+  cy.get(selectors.ValidatePolicyButton).click();
+});
+
+When(/^I test valid rego policy code$/, () => {
```

Comment:
Thoughts on merging this and the previous step? I don't know the name of the feature, but there's a way to get the value of `valid/invalid` passed to the step by using `{string}` (or whatever type) where that work would go: https://github.com/TheBrainFamily/cypress-cucumber-preprocessor#reusable-step-definitions

Reply:
I agree with merging. I tried to combining the steps before, but I had some issues with mocking the request/response, but I restructured some things so let me try again.

Reply:
JK was actually just a problem with my RegEx 🤦  Thanks for the suggestions!